USca/USca_ReportManager/Controls/ReportByTagId.xaml.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Windows;
+using System.Windows.Controls;
+using USca_ReportManager.Util;
+
+namespace USca_ReportManager.Controls
+{
+    public partial class ReportByTagId : UserControl, INotifyPropertyChanged
+    {
+        public ObservableCollection<TagLogDTO> TagLogs { get; set; } = new();
+        private TagLogService _tagLogService = new();
+        public string TagName { get; set; } = "";
+
+        public ReportByTagId()
+        {
+            InitializeComponent();
+        }
+
+        private async void BtnSearch_Click(object sender, RoutedEventArgs e)
+        {
+            string searchQuery = TxtTagID.Text;
+            if (!int.TryParse(searchQuery,  out int id))
+            {
+                MessageBox.Show("Please enter an integer!", "Failure", MessageBoxButton.OK);
+                return;
+            }
+
+            try
+            {
+                var res = await _tagLogService.GetByTag(id);
+                TagLogs.Clear();
+                foreach (var o in res.Logs)
+                {
+                    TagLogs.Add(o);
+                }
+                TagName = res.TagName;
+            }
+            catch (NotFoundException)
+            {
+                MessageBox.Show("Tag not found!", "Failure", MessageBoxButton.OK);
+                TagLogs.Clear();
+                TagName = "";
```

Comment:
First clear `TagLogs` and `TagName`, then display the message. Alternatively only show the message box without clearing old values, maybe replacing `TxtTagId.Text` with previous valid id? Doesn't really matter anyways.

Reply:
I did the first thing (swap order).
[89d2b85](https://github.com/magley/untitled-scada-project/pull/25/commits/89d2b8527281fc24569a356bb773d5c77aca992d)